### PR TITLE
Drop support for INITRD_MODULES (jsc#PED-1915)

### PR DESCRIPTION
**Pull request by mwilck**
- weak-modules2: ignore INITRD_MODULES (jsc#PED-1915)
- weak-modules2: obtain list of initrd modules from dracut (jsc#PED-1915)

### CODE REVIEW

**Comment by mwilck** [comment]
@aafeijoo-suse for information

File: weak-modules2 [hunk 1]
```diff
@@ -390,12 +390,72 @@ previous_version_of_kmp() {
...
+                    continue 2
+                fi
+            done
+        done
+    }
+}
+
+get_current_basenames() {
+    $LSINITRD /boot/initrd-$1 |
+	sed -rn 's:.*\<usr/lib/modules/.*/::p' |
```
Comment (aafeijoo-suse) [inline weak-modules2:424]: I think you accidentally removed two `\`, at the end of lines 423 and 424

File: weak-modules2 [hunk 2]
```diff
@@ -390,12 +390,72 @@ previous_version_of_kmp() {
...
+
+    # Assemble the content of the dracut configuration files.
+    # Make sure to put a newline between subsequent conf file contents.
+    for cf in $(dropindirs_sort .conf "$DRACUT_CONFDIR" "$DRACUT_BUILTIN_CONFDIR"); do
+	conf="$conf
+$(cat "$cf")"
+    done
+
+    # run get_dracut_drivers with reduced privileges
+    get_current_basenames "$1" |
```
Comment (aafeijoo-suse) [inline weak-modules2:454]: Is this pipe ok?

File: weak-modules2 (hunk 1 above)
Comment (aafeijoo-suse) [reply weak-modules2:424]: Oh, I see it's not necessary. Sorry for the noise.

File: weak-modules2 (hunk 2 above)
Comment (mwilck) [reply weak-modules2:454]: Why not?

Comment (aafeijoo-suse) [reply weak-modules2:454]: hmm... ok, the issue is in `get_dracut_drivers`, if there is no `omit_drivers` in any conf file, it does not print any modules.

Comment (mwilck) [reply weak-modules2:454]: Wow, good catch. Thanks.

Comment (mwilck) [reply weak-modules2:454]: Should be fixed with the fixup I just pushed.

Comment (aafeijoo-suse) [reply weak-modules2:454]: You're welcome. Yes, now it looks good.

**Comment by mwilck** [comment]
Rebased and squashed the fixup patch.

**Comment by mwilck** [comment]
@hramrach says he feels not familiar enough with the subject to review. So I guess we just move forward. Hardly anyone will be using `INITRD_MODULES` in factory anyway, I suppose.

**Pull request merged**